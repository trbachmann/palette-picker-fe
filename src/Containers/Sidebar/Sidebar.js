import React, { Component } from 'react';
import ProjectArea from '../ProjectArea/ProjectArea';
import { postProject } from '../../Thunks/postProject';
import { connect } from 'react-redux';

export class Sidebar extends Component{
  constructor() {
    super();
    this.state = {
      name: '',
      error: ''
    };
  };
  
  handleChange = (event) => {
    const { value } = event.target;
    this.setState({ name: value });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const { name } = this.state;
    const { projects } = this.props;
    const foundName = projects.find(project => {
      return name.toLowerCase() === project.name.toLowerCase()
    })
    if (foundName) {
      this.setState({error: 'That project name is already taken! Please enter a different one.'})
    } else {
      this.props.postProject(name);
      this.setState({name: '', error: ''});
    }
  };

  render() {
    const { error } = this.state;
    return(
      <div className='Sidebar--div'>
        <h2>My Projects</h2>
        { error && <p>{error}</p>}
        <form 
          autoComplete='off' 
          onSubmit={this.handleSubmit}>
          <input 
            placeholder='Enter New Project Name' 
            name='name' 
            value={this.state.name} 
            onChange={this.handleChange}/>
          <button>Create New Project</button>
        </form>
        <ProjectArea/>
      </div>
    )
  };
};

export const mapDispatchToProps = dispatch => ({
  postProject: (name) => dispatch(postProject(name))
});

export const mapStateToProps = state => ({
  projects: state.projects
})

export default connect(mapStateToProps, mapDispatchToProps)(Sidebar);