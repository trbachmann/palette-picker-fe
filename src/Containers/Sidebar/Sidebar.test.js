import React from 'react';
import { shallow } from 'enzyme';
import { Sidebar } from './Sidebar';

describe('Sidebar', () => {
  describe('Sidebar component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Sidebar />);
    });

    it('should match the snapshit', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});