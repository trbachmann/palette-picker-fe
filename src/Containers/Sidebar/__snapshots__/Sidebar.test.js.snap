// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Sidebar component should match the snapshot 1`] = `
<div
  className="Sidebar--div"
>
  <h2>
    My Projects
  </h2>
  <form
    autoComplete="off"
    onSubmit={[Function]}
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="Project Name"
      value=""
    />
    <button>
      Create New Project
    </button>
  </form>
  <Connect(ProjectArea) />
</div>
`;

exports[`Sidebar Sidebar component should set an error and match snapshot with error if the project name already exists 1`] = `
<div
  className="Sidebar--div"
>
  <h2>
    My Projects
  </h2>
  <p>
    That project name is already taken! Please enter a different one.
  </p>
  <form
    autoComplete="off"
    onSubmit={[Function]}
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="Project Name"
      value="Game Time"
    />
    <button>
      Create New Project
    </button>
  </form>
  <Connect(ProjectArea) />
</div>
`;
